source/darwin/securityframework_ecc.c

```diff
@@ -207,14 +230,46 @@ struct aws_ecc_key_pair *aws_ecc_key_pair_new_from_private_key_impl(
     enum aws_ecc_curve_name curve_name,
     const struct aws_byte_cursor *priv_key) {
 
-    struct aws_byte_cursor empty_cur;
-    AWS_ZERO_STRUCT(empty_cur);
+    /**
+     * We use SecCreateKeyWithData to create ECC key. Expected format for the key passed to that api is a byte buffer
+     * consisting of "0x04 | x | y | p", where x,y is public pair and p is private key.
+     *
+     * In this case we only have private key and we need to construct SecKey from that.
+     *
+     * We used to just pass 0,0 point for x,y, i.e. "0x04 | 0 | 0 | p".
+     *
+     * This used to work on Macs before 14, but in 14+ SecCreateKeyWithData returns error,
+     * which is reasonable since 0,0 is not a valid public point.
+     *
+     * To get around the issue, we use a fake public key, which is a valid public point, but not matching the private
+     * key as a quick workaround.
+     */
+    struct aws_byte_cursor fake_pub_x;
+    AWS_ZERO_STRUCT(fake_pub_x);
+    struct aws_byte_cursor fake_pub_y;
+    AWS_ZERO_STRUCT(fake_pub_y);
+    switch (curve_name) {
+        case AWS_CAL_ECDSA_P256:
+            fake_pub_x = aws_byte_cursor_from_array(FAKE_X_ECDSA_P256, sizeof(FAKE_X_ECDSA_P256));
+            fake_pub_y = aws_byte_cursor_from_array(FAKE_Y_ECDSA_P256, sizeof(FAKE_Y_ECDSA_P256));
+            break;
+        case AWS_CAL_ECDSA_P384:
+            fake_pub_x = aws_byte_cursor_from_array(FAKE_X_ECDSA_P384, sizeof(FAKE_X_ECDSA_P384));
+            fake_pub_y = aws_byte_cursor_from_array(FAKE_Y_ECDSA_P384, sizeof(FAKE_Y_ECDSA_P384));
+            break;
+
+        default:
+            break;
```

Comment:
raise error?

Reply:
we can fatal assert here? curve being supported should already have been verified

Reply:
I thought about that, but seems like our API doesn't really raise error for the unknown enum. But, yeah, I can raise the error here.